source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -236,11 +242,27 @@ Http::FilterTrailersStatus FaultFilter::decodeTrailers(Http::HeaderMap&) {
 }
 
 FaultFilterStats FaultFilterConfig::generateStats(const std::string& prefix, Stats::Scope& scope) {
-  std::string final_prefix = prefix + "fault.";
-  return {ALL_FAULT_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
+  const std::string final_prefix = prefix + "fault.";
+  return {ALL_FAULT_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix),
+                                 POOL_GAUGE_PREFIX(scope, final_prefix))};
 }
 
-void FaultFilter::onDestroy() { resetTimerState(); }
+void FaultFilter::incActiveFaults() {
+  // Only charge 1 active fault per filter in case we are injecting multiple faults.
+  if (fault_active_) {
+    return;
+  }
+
+  config_->stats().active_faults_.inc();
+  fault_active_ = true;
+}
+
+void FaultFilter::onDestroy() {
+  resetTimerState();
+  if (fault_active_) {
+    config_->stats().active_faults_.dec();
```

Comment:
hmm.. reading this, I am a bit more confused..

Reply:
What are you confused about? I can clarify.

Reply:
So starts are per HCM instance, i.e. one per thread per HCM? If so, then this makes sense..